src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -9,14 +9,36 @@
 #include <sys/stat.h>
 #include <linux/limits.h>
 
-#define MAX_NUM_TARGETS 15
-#define MAX_NUM_INSTALL_LINES 15
+#define MAX_NUM_TARGETS 48
+#define MAX_NUM_INSTALL_LINES 48
 #define MAX_NUM_UNITS 128
 #define MAX_BUF_SIZE 512
 
-static const char* UNIT_FILE_PREFIX = "/usr/lib/systemd/system/";
-static const char* CONFIG_FILE = "/etc/sonic/generated_services.conf";
-static const char* MACHINE_CONF_FILE = "/host/machine.conf";
+const char* UNIT_FILE_PREFIX = "/usr/lib/systemd/system/";
+const char* CONFIG_FILE = "/etc/sonic/generated_services.conf";
+const char* MACHINE_CONF_FILE = "/host/machine.conf";
+const char* ASIC_CONF_FORMAT = "/usr/share/sonic/device/%s/asic.conf";
+
+const char* g_unit_file_prefix;
+const char* get_unit_file_prefix() {
+    return (g_unit_file_prefix) ? g_unit_file_prefix : UNIT_FILE_PREFIX;
+}
+
+const char* g_config_file;
+const char* get_config_file() {
+    return (g_config_file) ? g_config_file : CONFIG_FILE;
+}
+
+const char* g_machine_config_file;
+const char* get_machine_config_file() {
+    return (g_machine_config_file) ? g_machine_config_file : MACHINE_CONF_FILE;
+}
+
+const char* g_asic_conf_format;
```

Comment:
Suggest explicitly setting setting to NULL in definition. Can't hurt to be explicit.